File: src/main/java/mil/nga/tiff/field/type/LongField.java
package mil.nga.tiff.field.type;

import java.nio.ByteBuffer;

abstract sealed class LongField<T extends Number> extends NumericFieldType<T> permits UnsignedLongField, SignedLongField {

    @Override
    final protected void writeSample(ByteBuffer buffer, T value) {
        buffer.putInt(value.intValue());
    }

    @Override
    final public void transferSample(ByteBuffer outBuffer, ByteBuffer inBuffer) {
        outBuffer.putInt(inBuffer.getInt());
    }

}

File: src/main/java/mil/nga/tiff/field/type/LongField.kt
package mil.nga.tiff.field.type

import java.nio.ByteBuffer

abstract class LongField<T : Number> : NumericFieldType<T>() {
    override fun writeSample(buffer: ByteBuffer, value: T) {
        buffer.putInt(value.toInt())
    }

    override fun transferSample(outBuffer: ByteBuffer, inBuffer: ByteBuffer) {
        outBuffer.putInt(inBuffer.getInt())
    }
}
